static/js/click_handlers.js

```diff
@@ -531,24 +532,35 @@ export function initialize() {
             $(".tooltip").remove();
         });
 
-    function do_render_buddy_list_tooltip(elem, title_data) {
-        let placement = "left";
-        let observer;
+    // Common properties of do_render_buddy_list_tooltip(elem, title_data)
+    // and do_render_status_emoji_name_tooltip(elem, title_data).
+    let placement;
+    const user_list_tooltip_props = {
+        delay: 0,
+        arrow: true,
+        allowHTML: true,
+        showOnCreate: true,
+    };
+
+    function calculate_tooltip_placement(placement) {
         if (window.innerWidth < media_breakpoints_num.md) {
             // On small devices display tooltips based on available space.
             // This will default to "bottom" placement for this tooltip.
             placement = "auto";
         }
+        return placement;
+    }
+
+    function do_render_buddy_list_tooltip(elem, title_data) {
+        placement = calculate_tooltip_placement("left");
+        let observer;
         tippy(elem[0], {
             // Quickly display and hide right sidebar tooltips
             // so that they don't stick and overlap with
             // each other.
-            delay: 0,
-            content: render_buddy_list_tooltip_content(title_data),
-            arrow: true,
+            ...user_list_tooltip_props,
             placement,
-            allowHTML: true,
-            showOnCreate: true,
+            content: render_buddy_list_tooltip_content(title_data),
             onHidden: (instance) => {
                 instance.destroy();
```

Comment:
We should also destroy `status_emoji_name_tooltip` instance if it is defined here. Since this causes the tooltip to stick around even if the emoji is no longer visible. Although, I am not able to reproduce this on this PR, this is something that happens definitely.

You can define `status_emoji_name_tooltip = instace` with `onShow` in `do_render_status_emoji_name_tooltip`.